src/dbnode/client/config.go

```diff
@@ -234,11 +236,10 @@ func (c Configuration) NewAdminClient(
 	writeRequestScope := iopts.MetricsScope().SubScope("write-req")
 	fetchRequestScope := iopts.MetricsScope().SubScope("fetch-req")
 
-	envCfg := environment.ConfigureResults{
-		TopologyInitializer: params.TopologyInitializer,
-	}
+	var envCfg environment.ConfigureResults
 
-	if envCfg.TopologyInitializer == nil {
+	// Initialize envCfg
+	if c.EnvironmentConfig != nil {
```

Comment:
it does HAVE to change, but I feel it is cleaner to change it.

The way it was written, it will not init/populate envCfg unless topoInit is not passed in, which is fine because topology Initializer is the only one that needed envCfg.

Now that this assumption does not hold anymore. 

It is cleaner to init envCfg from client.Configuration if EnviromentConfig is set (under the client section). For dbnode's admin client, client won't have EnviromentConfig set. For normal m3db client, client will have EnviromentConfig set.

Reply:
ok makes sense